File: GameStudies/CS/Ferramenta.cs
using UnityEngine;

public class Ferramenta
{
    public string Nome;
    public int Durabilidade = 100;
    public int Poder;

    public Ferramenta(string nome, int poder)
    {
        Nome = nome;
        Poder = poder;
    }

    public void Usar()
    {
        if (Durabilidade > 0)
        {
            Durabilidade--;
            Debug.Log(Nome + " usada! Durabilidade: " + Durabilidade);
        }
        else
        {
            Debug.Log(Nome + " está quebrada!");
        }
    }
}

File: GameStudies/CPP/Ferramenta.cpp
#include "Ferramenta.hpp"
#include <iostream>

Ferramenta::Ferramenta(std::string nome) {
    this->nome = nome;
    durabilidade = 100;
}

void Ferramenta::usar() {
    if (durabilidade > 0) {
        durabilidade--;
        std::cout << nome << " usada\n";
    }
}